### PR TITLE
refactor(common/models): factor out word breakers to their own package

### PR DESCRIPTION
**Pull request by eddieantonio**
Following up on #3119, this moves all of the word breakers to their own package: `@keymanapp/lexical-model-word-breakers`.

I also changed the tests to TypeScript and I follow a more conventional node/TypeScript directory structure.

Do we still want a `build.sh`? `npx lerna bootstrap`  from the repository root will install dev dependencies and build the TypeScript code, so that it's available in other modules.

### CODE REVIEW

**Comment by eddieantonio** [comment]
Hey @jahorton! Could you please look at the proposed code changes here?

It doesn't _quite_ work, and I believe that's due to the LMLayer requiring namespaces instead of commonjs packages. I think I'll need to convert this back to namespace'd packages in order to make it work ¯\\\_(ツ)\_/¯

But here's the proposed directory structure. It uses a more conventional NodeJS/TypeScript structure, and does _not_ require a `build.sh` to build and test:

```
.
├── README.md
├── lib/  # typescript compliation goes here
├── node_modules/ # omitted, to spare GitHub's server quota
├── package-lock.json
├── package.json
├── src
│   ├── ascii.ts
│   ├── default
│   │   ├── data.ts
│   │   └── index.ts
│   ├── index.ts
│   └── placeholder.ts
├── test
│   ├── test-ascii-word-breaker.ts
│   ├── test-default-word-breaker-with-trie.js
│   ├── test-default-word-breaker.ts
│   └── test-placeholder-word-breaker.ts
└── tsconfig.json
```

**Comment by jahorton** [comment]
Yeah, namespaces vs modules is extremely divisive with TS codebases.  I've raised issues within the team about this, as we'd be wise to convert to module use at some point, but unfortunately that won't be happening this version.

The directory structure and such sound fine.

**Comment by eddieantonio** [comment]
This one is ready! It creates the package `@keymanapp/models-wordbreakers`, which is included in the worker bundle in `common/predictive-text`.

File: common/models/wordbreakers/package.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{
+  "name": "@keymanapp/models-wordbreakers",
+  "version": "14.0.75",
+  "description": "Word breakers bundled in the LMlayer",
```
Comment (mcdurdin) [inline common/models/wordbreakers/package.json:4]:
```suggestion
  "description": "Word breakers bundled in the modeling layer",
```

File: common/models/wordbreakers/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# `@keymanapp/models-wordbreakers`
+
+The word breakers bundled in Keyman's lexical model layer (LMLayer).
```
Comment (mcdurdin) [inline common/models/wordbreakers/README.md:3]:
```suggestion
The word breakers bundled in Keyman's modeling layer (common/models).
```

**Pull request merged**